core/commands/add.go

```diff
@@ -113,12 +113,16 @@ remains to be implemented.
 					return
 				}
 
-				err = n.Pinning.Pin(context.Background(), rootnd, true)
+				rnk, err := rootnd.Key()
 				if err != nil {
 					res.SetError(err, cmds.ErrNormal)
 					return
 				}
 
+				mp := n.Pinning.GetManual()
+				mp.RemovePinWithMode(rnk, pin.Indirect)
```

Comment:
because it is added below (for all files found) and then re-added here?

why not just not pin anything until the end? i.e. remove the call to pin below and only pin recursively here after adding? 

is the worry that something will GC in-between, while adding?